### PR TITLE
Add cache events in bulk update func

### LINKED ISSUE #53: Add the PublishDocument event trigger after the bulk update
Need to add the PublishDocument trigger to the updateDocuments functions located in `database/postgresql/base.go:265` and `database/mongo/base.go:369`

It will require a new function, something that can grab multiple documents by ID to publish the changed documents they'll need to be fetched, to avoid the usage of the GetByID in a loop.

### PR DESCRIPTION
**Pull request by rostikts**
Added endpoint for getting several records by ids list as described in #53 

```
URL example: /db/tasks?byids=1
body: ['id1-here', 'id2-here', '...']
```

Also added the cache events in bulk update functions
Closes #53

### CODE REVIEW

File: db.go [hunk 1]
```diff
@@ -203,6 +205,42 @@ func (database *Database) query(w http.ResponseWriter, r *http.Request) {
...
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	col := ""
+
+	_, r.URL.Path = ShiftPath(r.URL.Path)
+	col, r.URL.Path = ShiftPath(r.URL.Path)
+
+	var body map[string][]string
```
Comment (dstpierre) [inline db.go:220]: I wonder if it'd be simpler to use a `[]string` directly where the caller of this API post `["id1", "id2"], ...]` and no need to wrap this in in an object. What do you think?

File: db.go [hunk 2]
```diff
@@ -203,6 +205,42 @@ func (database *Database) query(w http.ResponseWriter, r *http.Request) {
...
+	if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
```
Comment (dstpierre) [inline db.go:221]: I'd use the `parseBody` which handle closing the body as well.

File: db.go [hunk 3]
```diff
@@ -203,6 +205,42 @@ func (database *Database) query(w http.ResponseWriter, r *http.Request) {
...
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```
Comment (dstpierre) [inline db.go:222]: a Bad Request would be more precise here.

File: db.go [hunk 4]
```diff
@@ -203,6 +205,42 @@ func (database *Database) query(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+	ids, ok := body["ids"]
```
Comment (dstpierre) [inline db.go:225]: if you decide to go with the `[]string` we can remove this block completely.

**Review by dstpierre — COMMENTED** [review]
tiny comment regarding the type of data posted.

It's up-to-you to change or not.

File: db.go (hunk 1 above)
Comment (rostikts) [reply db.go]: Fixed

File: db.go (hunk 3 above)
Comment (rostikts) [reply db.go]: Fixed

File: db.go (hunk 2 above)
Comment (rostikts) [reply db.go]: Fixed

File: db.go (hunk 4 above)
Comment (rostikts) [reply db.go]: Fixed

**Pull request merged**